### PR TITLE
SAC implementation update

### LINKED ISSUE #211: Potential bug caused by calling policy.eval() before collecting experience
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [x] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```

In [this commit](https://github.com/thu-ml/tianshou/commit/8bb8ecba6ea28fdba8f6130755b0ddf493073c84) I found policy.eval() was called before collecting experience in both on/off policy trainer, but some algorithms, such as the DDPGPolicy Tianshou provided, use the attribute `training` to control exploration and exploitation. Wouldn't this change cause serious bug by preventing these algorithms from exploration? Also, I'm not sure if there were any other place that Tianshou uses `training` to distinguish training and testing.

piece of source code of DDPGPolicy:
```python
def forward(self, batch: Batch,
                state: Optional[Union[dict, Batch, np.ndarray]] = None,
                model: str = 'actor',
                input: str = 'obs',
                explorating: bool = True,
                **kwargs) -> Batch:
        """Compute action over the given batch data.
        :return: A :class:`~tianshou.data.Batch` which has 2 keys:
            * ``act`` the action.
            * ``state`` the hidden state.
        .. seealso::
            Please refer to :meth:`~tianshou.policy.BasePolicy.forward` for
            more detailed explanation.
        """
        model = getattr(self, model)
        obs = getattr(batch, input)
        actions, h = model(obs, state=state, info=batch.info)
        actions += self._action_bias
        if self.training and explorating:
            actions += to_torch_as(self._noise(actions.shape), actions)
        actions = actions.clamp(self._range[0], self._range[1])
        return Batch(act=actions, state=h)
```

### PR DESCRIPTION
**Pull request by danagi**
- replace DiagGuassian with Independent(Normal) (pytorch has already supported this)
- detach alpha from autograd
- add value/alpha to result (more informational)
- revert #204 to fix #211

### CODE REVIEW

**Comment by Trinkle23897** [comment]
Could you please have a try at `examples/box2d/bipedal_hardcore_sac.py` with auto-alpha tuning? Thanks very much~

**Review by duburcqa — DISMISSED** [review]

File: examples/box2d/bipedal_hardcore_sac.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ def get_args():
     parser.add_argument('--gamma', type=float, default=0.99)
     parser.add_argument('--tau', type=float, default=0.005)
     parser.add_argument('--alpha', type=float, default=0.1)
+    parser.add_argument('--auto_alpha', type=bool, default=True)
+    parser.add_argument('--alpha_lr', type=bool, default=3e-4)
```
Comment (Trinkle23897) [inline examples/box2d/bipedal_hardcore_sac.py]: do not bool

**Review by Trinkle23897 — APPROVED** [review]
